pkg/integrations/v4/dm/emitter.go

```diff
@@ -149,16 +152,28 @@ func (e *emitter) runFwReqConsumer(ctx context.Context) {
 
 		case req := <-e.reqsQueue:
 			e.measure(instrumentation.Counter, instrumentation.DMDatasetsReceived, int64(len(req.Data.DataSets)))
+
+		loop:
 			for _, ds := range req.Data.DataSets {
 				select {
 				case _ = <-ctx.Done():
 					return
 				default:
 					if ds.IgnoreEntity {
-						// Should not contain entity attributes
-						e.processDatasetNoRegister(req.Data.Integration, req.FwRequestMeta, ds)
-					} else {
+						e.emidDatasetWithEmptyEntity(req.Data.Integration, req.FwRequestMeta, ds)
+						continue loop //nolint:nlreturn
+					}
+					if ds.Entity.IsAgent() {
+						e.emitDatasetForAgent(ctx, req.Data.Integration, req.FwRequestMeta, ds)
+						continue loop //nolint:nlreturn
+					}
+					deprecateRegisterFFEnabled, deprecateRegisterFFExists := e.ffRetriever.GetFeatureFlag(fflag.FlagDmRegisterDeprecated)
+					if !deprecateRegisterFFExists || (deprecateRegisterFFExists && !deprecateRegisterFFEnabled) {
 						e.processDatasetRegister(ctx, req.Data.Integration, req.FwRequestMeta, ds)
+						continue loop //nolint:nlreturn
+					}
+					if deprecateRegisterFFExists && deprecateRegisterFFEnabled {
+						elog.Warn("Register for DM integrations is deprecated.")
```

Comment:
What do you think if we add some tags in the log entry to easily identify the corresponding integration? In addition, I would add a bit more context to make the user aware that this integration data won't be forwarded.

Reply:
good point. Adding it

Reply:
Improved message in https://github.com/newrelic/infrastructure-agent/pull/1357/commits/1594a405dae06536d4e70430100dc4cc58f802d7